### PR TITLE
Added mart to monitor NPQ applications where the approval status has reversed from accepted to pending.

### PR DESCRIPTION
**Pull request by AntPage1**

### CODE REVIEW

File: definitions/marts/looker_studio_marts/npq/ls_npq_acceptance_reversal.sqlx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    The volume of applications where this activity has occurred can be aggregated at cohort, lead provider, course and current application status levels.",
+    columns: {
+        current_application_status: "The current application status as per the service.",
+        accepted_reversal_count: "The number of times the application status of the application has been reverted from accepted to pending."
+    }
+}
+
+WITH acceptance_reversion AS (
+  SELECT
+    status_changes.entity_id AS application_ecf_id
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/npq/ls_npq_acceptance_reversal.sqlx:20]: Apart from the `application_ecf_id`, is there any reason why you need to group by the other fields at this level? Could you not just group by at this id level?

File: definitions/marts/looker_studio_marts/npq/ls_npq_acceptance_reversal.sqlx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    lp.id = status_changes.npq_lead_provider_id
+  LEFT JOIN
+    ${ref('cohorts_latest_cpd')} AS cohorts
+  ON
+    cohorts.id = status_changes.cohort_id
+  WHERE
+    status_changes.key_updated = 'lead_provider_approval_status'
+    AND
+    status_changes.previous_value = 'accepted'
+  GROUP BY ALL
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/npq/ls_npq_acceptance_reversal.sqlx:43]: I like this! (I haven't seen it before)

**Review by consult1815 — CHANGES_REQUESTED** [review]

File: definitions/marts/looker_studio_marts/npq/ls_npq_acceptance_reversal.sqlx (hunk 1 above)
Comment (AntPage1) [reply definitions/marts/looker_studio_marts/npq/ls_npq_acceptance_reversal.sqlx:20]: The group by all does two things in one go. It groups the count and also removes duplicated rows without the need of an additional qualify based on update occurred at date.

**Review by consult1815 — APPROVED** [review]
Looks good

**Pull request merged**